src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py

```diff
@@ -7007,6 +7007,61 @@ def test_node_public_ip_tags(self, resource_group, resource_group_location):
             self.is_empty(),
         ])
 
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='eastus')
+    def test_vms_agentpool_type(self, resource_group, resource_group_location):
+        aks_name = self.create_random_name('cliakstest', 16)
+        nodepool_name = self.create_random_name('n', 6)
+        nodepool_name_1 = self.create_random_name('n', 6)
+
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': aks_name,
+            'location': resource_group_location,
+            'ssh_key_value': self.generate_ssh_keys(),
+            'node_pool_name': nodepool_name,
+            'node_vm_size': 'standard_d2a_v4',
+        })
+
+        self.cmd(
+            'aks create '
+            '--resource-group={resource_group} '
+            '--name={name} '
+            '--location={location} '
+            '--ssh-key-value={ssh_key_value} '
+            '--nodepool-name={node_pool_name} '
+            '--node-count=1 '
+            '--node-vm-size={node_vm_size} '
+            '--vm-set-type="VirtualMachines"',
```

Comment:
Is the usage of this vm type guarded by something like AFEC feature flag.

Reply:
right, we have feature flag called "Microsoft.ContainerService/VMsAgentPoolPreview"

Reply:
Seems we have the feature flag, but missed the validation at RP. I will add the code there.

Reply:
actually, we have toggle to enable this feature, right now only our e2e and 4 test subscriptions enabled this.